rotkehlchen/chain/ethereum/modules/yearn/utils.py

```diff
@@ -0,0 +1,129 @@
+import logging
+from json import JSONDecodeError
+from typing import TYPE_CHECKING
+
+import requests
+
+from rotkehlchen.assets.asset import UnderlyingToken
+from rotkehlchen.assets.utils import get_or_create_evm_token
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.misc import ONE
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import (
+    YEARN_VAULTS_V1_PROTOCOL,
+    YEARN_VAULTS_V2_PROTOCOL,
+    ChainID,
+    EvmTokenKind,
+    GeneralCacheType,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+YEARN_OLD_API = 'https://api.yearn.finance/v1/chains/1/vaults/all'
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_yearn_vaults(db: 'DBHandler') -> None:
+    """Query yearn API and ensure that all the tokens exist locally. If they exist but the protocol
+    is not the correct one, then the asset will be edited.
+
+    May raise:
+    - RemoteError
+    """
+    try:
+        response = requests.get(YEARN_OLD_API)
+        data = response.json()
+    except requests.exceptions.RequestException as e:
+        raise RemoteError(f'Failed to obtain yearn vault information. {str(e)}') from e
+    except (DeserializationError, JSONDecodeError) as e:
+        raise RemoteError(f"Failed to deserialize data from yearn's old api. {str(e)}") from e
+
+    if not isinstance(data, list):
+        raise RemoteError(f'Unexpected format from yearn vaults reponse. Expected a list, got {data}')  # noqa: E501
+
+    num_of_vaults = GlobalDBHandler().get_general_cache_values(
```

Comment:
This is a **terrible** variable name. You call it "number of vaults" which any logical reader will automatically assume is an `int`.

But it's not right? I had to go and check the code to see wtf `get_general_cache_values()` does. It's a `list[str]`.

Which really really really confuses me. I don't understand what it is supposed to return, and especially don't understand what your if condition checks for below. Please fix this and make it readable.